### DISCUSSION TITLE
Defining a pass-through type

### DISCUSSION THREAD

**Discussion by patrick91** [python/typing#954] [Q&A] [answered]
Hi folks, here I am with another question 😊 Excuse me for the title, not sure it makes sense :)

In Strawberry we have a class called `Private` that can be used to annotate some fields, it works similarly to 
dataclasses' InitVar, here's an example:

```python
import strawberry

@strawberry.type
class User:
    name: str
    age: strawberry.Private[int]

patrick = User(name="Patrick", age=1)
reveal_type(patrick.age)
```

unfortunately this breaks the type checkers since `patrick.age` is Private :)

For MyPy we made a plugin to prevent this problem, is there a way to make this class a pass through to the passed type?

The only alternative I'm thinking of is to convert the class to be a function that returns `typing.Annotated`, but I wonder if there's a better alternative :)

Here's our implementation:

```python
class Private:
    __slots__ = ("type",)

    def __init__(self, type):
        self.type = type

    def __repr__(self):
        if isinstance(self.type, type):
            type_name = self.type.__name__
        else:
            # typing objects, e.g. List[int]
            type_name = repr(self.type)
        return f"strawberry.Private[{type_name}]"

    def __class_getitem__(cls, type):
        return Private(type)
```

**Comment by erictraut** [comment]
Another option is to adopt something similar to the dataclass `field` descriptor function. This is a pattern that `attrs`, `pydantic` and other libraries use as well, so it's a well-trodden path.

It would look something like this:
```python
@strawberry.type
class User:
    name: str
    age: int = strawberry.field(private=True)
```

(One potential problem with `strawberry.field` is that it might invite a copyright infringement lawsuit from the Beatles. ;) )

This approach would also work with [dataclass transform](https://github.com/microsoft/pyright/blob/main/specs/dataclass_transforms.md).

**Reply by patrick91** [reply]
@erictraut thanks for the answer! we already have `strawberry.field` but we'd like to have a different syntax, I'll experiment with typing.Annotated and see if I can get something working ^^

**Reply by patrick91** [reply]
I solved this with annotated and getting rid of the custom type!

**Comment by patrick91** [comment, marked as answer]
I solved with `Private = Annotated[T, StrawberryPrivate()]` ^^

https://github.com/strawberry-graphql/strawberry/pull/1437/files
